copying falsifying example into `@example()` fails to reproduce test failure due to pprinter sorting dict keys

Some code depends on the order of dict iteration so sorting the keys before printing makes reproducing a failing test more difficult. 

Related to #2913 ?

**Sample code**:
```
import hypothesis as h
import hypothesis.strategies as st

@h.given(st.dictionaries(st.integers(), st.integers()))
def test(d):
    h.note(f"{d=}")
    assert sorted(d) == list(d)
```

**Output**: falsifying example report gives an input that if copied verbatim, would actually pass the test
```
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.8.11, pytest-5.4.3, py-1.11.0, pluggy-0.13.1
rootdir: [snip]/hypothesis-dictionary-order
plugins: hypothesis-6.47.0
collected 1 item                                                                                                                                                                                          

test.py F                                                                                                                                                                                           [100%]

================================================================================================ FAILURES =================================================================================================
__________________________________________________________________________________________________ test ___________________________________________________________________________________________________

    @h.given(st.dictionaries(st.integers(), st.integers()))
>   def test(d):

test.py:5: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

d = {0: 0, 1: 0}

    @h.given(st.dictionaries(st.integers(), st.integers()))
    def test(d):
        h.note(f"{d=}")
>       assert sorted(d) == list(d)
E       assert [0, 1] == [1, 0]
E         At index 0 diff: 0 != 1
E         Use -v to get the full diff

test.py:7: AssertionError
----------------------------------------------------------------------------------------------- Hypothesis ------------------------------------------------------------------------------------------------
Falsifying example: test(
    d={0: 0, 1: 0},
)
d={1: 0, 0: 0}
========================================================================================= short test summary info =========================================================================================
FAILED test.py::test - assert [0, 1] == [1, 0]
```

**Expected output**: falsifying example should reflect the dict key order as actually passed into the test function

--- a/AUTHORS.rst
+++ b/AUTHORS.rst
@@ -32,6 +32,7 @@ their individual contributions.
 * `Chase Garner <https://www.github.com/chasegarner>`_ (chase@garner.red)
 * `Cheuk Ting Ho <https://github.com/Cheukting>`_
 * `Chris Down  <https://chrisdown.name>`_
+* `Chris van Dronkelaar <https://github.com/Chrisvandr>`_
 * `Christopher Martin <https://www.github.com/chris-martin>`_ (ch.martin@gmail.com)
 * `Claudio Jolowicz <https://github.com/cjolowicz>`_
 * `Conrad Ho <https://www.github.com/conradho>`_ (conrad.alwin.ho@gmail.com)
@@ -126,6 +127,7 @@ their individual contributions.
 * `Pierre-Jean Campigotto <https://github.com/PJCampi>`_
 * `Przemek Konopko <https://github.com/soutys>`_
 * `Richard Boulton <https://www.github.com/rboulton>`_ (richard@tartarus.org)
+* `Richard Scholtens <https://github.com/richardscholtens>`_ (richardscholtens2@gmail.com)
 * `Robert Howlett <https://github.com/jebob>`_
 * `Robert Knight <https://github.com/robertknight>`_ (robertknight@gmail.com)
 * `Rónán Carrigan <https://www.github.com/rcarriga>`_ (rcarriga@tcd.ie)
@@ -138,6 +140,7 @@ their individual contributions.
 * `Sangarshanan <https://www.github.com/sangarshanan>`_ (sangarshanan1998@gmail.com)
 * `Sanyam Khurana <https://github.com/CuriousLearner>`_
 * `Saul Shanabrook <https://www.github.com/saulshanabrook>`_ (s.shanabrook@gmail.com)
+* `Sebastiaan Zeeff <https://github.com/SebastiaanZ>`_ (sebastiaan.zeeff@ordina.nl)
 * `Shlok Gandhi <https://github.com/shlok57>`_ (shlok.gandhi@gmail.com)
 * `Sogata Ray <https://github.com/rayardinanda>`_ (rayardinanda@gmail.com)
 * `Stuart Cook <https://www.github.com/Zalathar>`_
@@ -155,6 +158,7 @@ their individual contributions.
 * `Tyler Nickerson <https://www.github.com/nmbrgts>`_
 * `Vidya Rani <https://www.github.com/vidyarani-dg>`_ (vidyarani.d.g@gmail.com)
 * `Vincent Michel <https://www.github.com/vxgmichel>`_ (vxgmichel@gmail.com)
+* `Viorel Pluta <https://github.com/viopl>`_ (viopluta@gmail.com)
 * `Vytautas Strimaitis <https://www.github.com/vstrimaitis>`_
 * `Will Hall <https://www.github.com/wrhall>`_ (wrsh07@gmail.com)
 * `Will Thompson <https://www.github.com/wjt>`_ (will@willthompson.co.uk)

--- a/hypothesis-python/RELEASE.rst
+++ b/hypothesis-python/RELEASE.rst
@@ -1,0 +1,8 @@
+RELEASE_TYPE: patch
+
+
+Order of dictionaries is now enforced in Python 3.7 and higher. This means that the original order of dictionaries matters for falsifying examples.
+
+This PR closes #3370.
+
+This PR was kindly supported by Ordina Pythoneers.

--- a/hypothesis-python/RELEASE.rst
+++ b/hypothesis-python/RELEASE.rst
@@ -1,8 +1,6 @@
 RELEASE_TYPE: patch
 
-
-Order of dictionaries is now enforced in Python 3.7 and higher. This means that the original order of dictionaries matters for falsifying examples.
-
-This PR closes #3370.
-
-This PR was kindly supported by Ordina Pythoneers.
+Our pretty-printer no longer sorts dictionary keys, since iteration order is
+stable in Python 3.7+ and this can affect reproducing examples (:issue:`3370`).
+This PR was kindly supported by `Ordina Pythoneers
+<https://www.ordina.nl/vakgebieden/python/>`__.

--- a/hypothesis-python/tests/cover/test_pretty.py
+++ b/hypothesis-python/tests/cover/test_pretty.py
@@ -134,6 +134,7 @@ def test_list():
 def test_dict():
     assert pretty.pretty({}) == "{}"
     assert pretty.pretty({1: 1}) == "{1: 1}"
+    assert pretty.pretty({1: 1, 0: 0}) == "{1: 1, 0: 0}"
 
 
 def test_tuple():

